packages/roosterjs-editor-api/lib/format/getFormatState.ts

```diff
@@ -1,45 +1,101 @@
+ */
+export function getPendableFormatState(document: Document): PendableFormatState {
+    let keys = Object.keys(PendableFormatCommandMap) as PendableFormatNames[];
+
+    return keys.reduce(
+        (state, key) => {
+            state[key] = document.queryCommandState(PendableFormatCommandMap[key]);
+            return state;
+        },
+        <PendableFormatState>{}
+    );
+}
+
+/**
+ * Get element based Format State at cursor
+ * @param editor The editor instance
+ * @param event (Optional) The plugin event, it stores the event cached data for looking up.
  * In this function the event cache is used to get list state and header level. If not passed,
  * it will query the node within selection to get the info
- * @returns The format state at cursor
+ * @returns An ElementBasedFormatState object
  */
-export default function getFormatState(editor: Editor, event?: PluginEvent): FormatState {
-    let range = editor.getSelectionRange();
-    let node = range && Position.getStart(range).normalize().node;
-    let styles = node ? getComputedStyles(node) : [];
+export function getElementBasedFormatState(
+    editor: Editor,
+    event?: PluginEvent
+): ElementBasedFormatState {
     let listTag = getTagOfNode(cacheGetElementAtCursor(editor, event, 'OL,UL'));
     let headerTag = getTagOfNode(cacheGetElementAtCursor(editor, event, 'H1,H2,H3,H4,H5,H6'));
-    let document = editor.getDocument();
-    return {
-        fontName: styles[0],
-        fontSize: styles[1],
-        textColor: styles[2],
-        backgroundColor: styles[3],
-
-        isBold: document.queryCommandState(DocumentCommand.Bold),
-        isItalic: document.queryCommandState(DocumentCommand.Italic),
-        isUnderline: document.queryCommandState(DocumentCommand.Underline),
-        isStrikeThrough: document.queryCommandState(DocumentCommand.StrikeThrough),
-        isSubscript: document.queryCommandState(DocumentCommand.Subscript),
-        isSuperscript: document.queryCommandState(DocumentCommand.Superscript),
 
+    return {
         isBullet: listTag == 'UL',
         isNumbering: listTag == 'OL',
         headerLevel: (headerTag && parseInt(headerTag[1])) || 0,
 
         canUnlink: !!editor.queryElements('a[href]', QueryScope.OnSelection)[0],
         canAddImageAltText: !!editor.queryElements('img', QueryScope.OnSelection)[0],
         isBlockQuote: !!editor.queryElements('blockquote', QueryScope.OnSelection)[0],
+    };
+}
 
+/**
+ * Get style based Format State at cursor
+ * @param editor The editor instance
+ * @returns A StyleBasedFormatState object
+ */
+export function getStyleBasedFormatState(editor: Editor): StyleBasedFormatState {
+    let range = editor.getSelectionRange();
+    let node = range && Position.getStart(range).normalize().node;
+    let styles = node ? getComputedStyles(node) : [];
+    return {
+        fontName: styles[0],
+        fontSize: styles[1],
+        textColor: styles[2],
+        backgroundColor: styles[3],

```

Comment:
What happens when styles is []?

Reply:
it can't be [], you can read the code of getComputedStyles()

Reply:
I meant if node is null (line 75)

Reply:
Oh I see. In that case it will just return a bunch of "undefined" which should be expected, also the original behavior.